app/models/network_map.rb

```diff
@@ -191,13 +190,24 @@ def display_title(lock: false)
 
   # Editor methods
   # These are only for oligrapher version 3
+  def confirmed_editor_ids
+    editors.filter { |e| !e.pending }.map(&:id)
+  end
+
+  def pending_editor_ids
+    editors.filter { |e| e.pending }.map(&:id)
+  end
+
+  def all_editor_ids
+    editors.map(&:id)
+  end
 
   def add_editor(editor)
     editor_id = editor.try(:id) || editor.try!(:to_i)
-    return self if editors.include?(editor_id)
+    return self if all_editor_ids.include?(editor_id)
 
     if validate_editor(editor_id)
-      editors << editor_id
+      editors << OpenStruct.new({ id: editor_id, pending: true })
```

Comment:
Is there a particular reason you used an OpenStruct here instead of a regular struct or a hash? OpenStructs are useful when the fields are dynamic, but in this case,  each editor should have the same fields, so I'd think a regular hash or defined struct would be better suited.

Reply:
No particular reason other than a hunch that accessor methods would save me time and be more readable and require less rewriting if we eventually wanted to create an editors table.